### PR TITLE
Changed language of email messages, now they are based of user setting.

### PR DESCRIPTION
**Pull request by MrK191**

### CODE REVIEW

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/mail/MailMessageCreator.java [hunk 1]
```diff
@@ -447,15 +550,23 @@ private String getSubjectTextForDatasetVersion(NotificationType notificationType
...
+                                                      getCurrentUser().getNotificationsLanguage(),
+                                                      Collections.singletonList(rootDataverseName));
             }
         }
 
         return StringUtils.EMPTY;
     }
 
+    private AuthenticatedUser getCurrentUser() {
+        return (AuthenticatedUser) session.getUser();
```
Comment (madryk) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/mail/MailMessageCreator.java]: The whole point was to use language of user that will receive the email - not language of the user that initiates the action

**Review by madryk — CHANGES_REQUESTED** [review]

File: dataverse-persistence/src/main/resources/Bundle_en.properties [hunk 2]
```diff
@@ -647,7 +647,7 @@ notification.email.requestFileAccess=File access requested for dataset: {0} by {
 notification.email.grantFileAccess=Access granted for files in dataset: {0} (view at {1}).
 notification.email.rejectFileAccess=Your request for access was rejected for the requested files in the dataset: {0} (view at {1}). If you have any questions about why your request was rejected, you may reach the dataset owner using the "Contact" link on the upper right corner of the dataset page.
 # Bundle file editors, please note that "notification.email.createDataverse" is used in a unit test
-notification.email.createDataverse=Your new dataverse named {0} (view at {1} ) was created in {2} (view at {3} ). To learn more about what you can do with your dataverse, check out the Dataverse Management - User Guide at {4}/{5}/user/dataverse-management.html .
+notification.email.createDataverse=Your new dataverse named {0} (view at {1} ) was created in {2} (view at {3}). To learn more about what you can do with your dataverse, check out the Dataverse Management - User Guide at {4}/{5}/user/dataverse-management.html .
```
Comment (wtr-icm) [inline dataverse-persistence/src/main/resources/Bundle_en.properties]: Do we want to preserve this space between `{1}` and `)`?

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/mail/MailMessageCreator.java [hunk 3]
```diff
@@ -403,53 +475,89 @@ private String authenticatedUserMessage(EmailNotificationDto notificationDto, St
...
 
-    private String getSubjectText(NotificationType notificationType, String rootDataverseName) {
+    private String getSubjectText(EmailNotificationDto notificationDto, String rootDataverseName) {
         List<String> rootDvNameAsList = Collections.singletonList(rootDataverseName);
-        switch (notificationType) {
+        switch (notificationDto.getNotificationType()) {
             case ASSIGNROLE:
-                return BundleUtil.getStringFromBundle("notification.email.assign.role.subject", rootDvNameAsList);
+                return BundleUtil.getStringFromBundle("notification.email.assign.role.subject",
+                                                      notificationDto.getNotificationReceiver().getNotificationsLanguage(),
```
Comment (wtr-icm) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/mail/MailMessageCreator.java]: Maybe we can put this `notificationDto.getNotificationReceiver().getNotificationsLanguage()` into some variable, and reduce repetitions in following lines?

**Review by wtr-icm — APPROVED** [review]

File: dataverse-persistence/src/main/resources/Bundle_en.properties (hunk 2 above)
Comment (MrK191) [reply dataverse-persistence/src/main/resources/Bundle_en.properties]: Hmm, I will ask @wfenrich about it.

**Review by wtr-icm — APPROVED** [review]

**Review by madryk — APPROVED** [review]

**Pull request merged**